app/controllers/events_controller.rb

```diff
@@ -0,0 +1,40 @@
+class EventsController < ApplicationController
+  helper_method :upcoming, :past
+  def new
+    @event = Event.new
+  end
+
+  def index
+    @events = Event.all
+  end
+
+  def show
+    @event = Event.find(params[:id])
+    @user = User.find(@event.creator_id)
+  end
+
+  def past
+    Event.past
+  end
+
+  def upcoming
+    Event.upcoming
+  end
```

Comment:
are these necessary? It'd be better to have `@upcoming = Event.upcoming` etc. in `index`